compiler/rustc_ast/src/ast.rs

```diff
@@ -1874,11 +1874,13 @@ impl StrLit {
 #[derive(HashStable_Generic)]
 pub enum LitIntType {
     /// e.g. `42_i32`.
-    Signed(IntTy),
+    /// The bool signals whether the literal is negated
+    Signed(IntTy, bool),
```

Comment:
I think it's worth a `enum Sign { Pos, Neg }` instead of the `bool`, much like we have `enum Mutability { Not, Mut }`.

Reply:
`Sign::Pos` is somewhat misleading, since there is 0. Should it be `NonNeg`?